internal/cloud/metadata/metadata.go

```diff
@@ -43,21 +39,3 @@ type InstanceLister interface {
 	// List retrieves all instances belonging to the current constellation.
 	List(ctx context.Context) ([]InstanceMetadata, error)
 }
-
-// JoinServiceEndpoints returns the list of endpoints for the join service, which are running on the control plane nodes.
-func JoinServiceEndpoints(ctx context.Context, lister InstanceLister) ([]string, error) {
```

Comment:
Since `joinclient.go` and `rejoinclient.go` have to perform the same action to fetch loadbalancer endpoint, and control plane IPs, what about adapting this function to include fetching of LB endpoint and using it for both?
With the current implementation, the code is slightly different for `joinclient.go` and `rejoinclient.go`.

Function could look like the following:
```Go
type endpointLister interface {
    List(ctx context.Context) ([]InstanceMetadata, error)
    GetLoadBalancerEndpoint(ctx context.Context) (host, port string, err error)
}

func JoinServiceEndpoints(ctx context.Context, lister endpointLister) ([]string, error) {
    lbIP, _, err := lister.GetLoadBalancerEndpoint(ctx)
    if err != nil {
        // Maybe do additional error handling here in case of no loadbalancer (ignore error?)
        return nil, fmt.Errorf("retrieving load balancer endpoint from cloud provider: %w", err)
    }
    // Old JoinServiceEndpoints code
    joinEndpoints := []string{}
    ...

    // Return all endpoints, with LB as the first endpoint
    return append(net.JoinHostPort(lbEndpoint, strconv.Itoa(constants.JoinServiceNodePort)), joinEndpoints), nil
}
```

Reply:
Hm, I didn't like the place of this function as part of the metadata package as it is too specific to me. Also, without this function the metadata package looks cleaner. 
I see you point of introducing a potential difference in both implementation though, but I haven't found the right spot.